src/module/entities/helper.js

```diff
@@ -841,6 +841,64 @@ module.exports = class UserProjectsHelper {
 			}
 		})
 	}
+
+	/**
+	 * Fetches entity documents based on entity type.
+	 * @method
+	 * @name entityListBasedOnEntityType
+	 * @param {string} type - Type of entity to fetch documents for.
+	 * @returns {Promise<Object>} Promise that resolves with fetched documents or rejects with an error.
+	 */
+
+	static entityListBasedOnEntityType(type) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				// Fetch the list of entity types available
+				const entityList = await entityTypeQueries.entityTypesDocument(
+					{
+						name: type,
+					},
+					['name']
+				)
+				// Check if entity list is empty
+				if (entityList.length < 0) {
+					throw {
+						status: HTTP_STATUS_CODE.not_found.status,
+						message: CONSTANTS.apiResponses.ENTITYTYPE_NOT_FOUND,
+					}
+				}
+				const projection = ['_id', 'metaInformation.name']
+				// Fetch documents for the matching entity type
+				let fetchList = await entitiesQueries.entityDocuments(
+					{
+						entityType: type,
+					},
+					projection
+				)
+
+				// Transform the fetched list to match the required result format
+				const result = fetchList.map((entity) => ({
+					_id: entity._id,
+					name: entity.metaInformation.name,
+				}))
+
+				if (fetchList.length < 1) {
```

Comment:
@MallanagoudaB you can do the check before the map function and condition !fetchList>0